vizro-core/src/vizro/models/types.py

```diff
@@ -299,10 +353,30 @@ def _parse_json(
             ) from exc
 
         import_path = json_schema_extra["import_path"]
-        try:
-            function = getattr(importlib.import_module(import_path), function_name)
-        except (AttributeError, ModuleNotFoundError) as exc:
-            raise ValueError(f"_target_={function_name} cannot be imported from {import_path}.") from exc
+
+        # Try multiple approaches to import the function - order matters here.
+        import_attempts = [
+            import_path + ":" + function_name,
+            function_name,
+        ]
+
+        for import_string in import_attempts:
+            try:
+                function: CapturedCallable = TypeAdapter(ImportString).validate_python(import_string)
+                break
+            except ValidationError:
+                continue
+        else:
+            if function_name in callable_defs:
+                return CapturedCallableProxy(
+                    function_name=function_name,
```

Comment:
Given this could now be `something:function_name` probably we should rename the variable since it's not just `function_name` any more.
```suggestion
                    _target_=_target_,
```

Maybe? Weird name but it makes it clear exactly what this is. I'd rename from L295 onwards.